File: src/main/java/de/cubbossa/pathfinder/Named.java
package de.cubbossa.pathfinder;

import net.kyori.adventure.text.Component;

/**
 * An object that can be represented as Component.
 */
public interface Named {

	/**
	 * @return The display name formatted in the MiniMessage style.
	 * @see "https://docs.adventure.kyori.net/minimessage/format.html"
	 */
	String getNameFormat();

	/**
	 * Sets the display name. The call if this method must also refresh the display name component if stored locally.
	 *
	 * @param name The display name formatted in the MiniMessage style.
	 * @see "https://docs.adventure.kyori.net/minimessage/format.html"
	 */
	void setNameFormat(String name);

	/**
	 * @return The display name as kyori Component. Must be the parsed version of the name format string and
	 * may be cached.
	 */
	Component getDisplayName();
}

File: pathfinder-api/src/main/java/de/cubbossa/pathfinder/misc/Named.kt
package de.cubbossa.pathfinder.misc

import net.kyori.adventure.text.Component

/**
 * An object that can be represented as Component.
 */
interface Named {
    /**
     * @return The display name formatted in the MiniMessage style.
     * @see "https://docs.adventure.kyori.net/minimessage/format.html"
     */
    val nameFormat: String

    /**
     * @return The display name as kyori Component. Must be the parsed version of the name format string and
     * may be cached.
     */
    val displayName: Component
}
